### DISCUSSION TITLE
GeoJsonLayer icon auto packing icon weirdly resized

### DISCUSSION THREAD

**Discussion by radoslavirha** [visgl/deck.gl#7832] [Q&A]
Hi,

I have GeoJsonLayer with icons using auto packing like:

```js
{
  url: iconUrl,
  height: 100,
  width: 100
}
```

which is fine if width:height is 1:1.

The problem is with icons with different aspect ratio. E.g. I have an icon 25x30px

https://user-images.githubusercontent.com/23722745/231797230-d42e3199-03c6-49f1-92db-2a1e2f90c6af.mov

On map init it looks fine, but when I zoom/move map, the icon is cut or resized.

I really don't know width/height because iconUrl is dynamic. Can it be solved on deck.gl side?

Thanks

**Comment by Pessimistress** [comment]
A change was made in 8.9 to preserve the aspect ratio of auto-packed icons: https://github.com/visgl/deck.gl/pull/7651

I think (without seeing a live demo) the problem you are experiencing here is that the icon gets scaled up to fit 100x100, and drawn right against the edge of the bounding box. Some interpolation artifacts are introduced in the process.

A better default behavior for unknown icon dimensions might be:
- Only scale down, never scale up; or
- Only scale by integer factors
